crates/gen/src/gen/mod.rs

```diff
@@ -0,0 +1,33 @@
+mod class;
+mod debug;
+mod delegate;
+mod r#enum;
+mod futures;
+mod interface;
+mod iterator;
+mod method;
+mod namespace;
+mod param;
+mod required_interface;
+mod r#struct;
+mod r#type;
+mod type_guid;
+mod type_kind;
+mod type_name;
+
+pub use class::*;
```

Comment:
I love the move to a `gen` submodule, but I also really wish we didn't rely so much on `*` glob imports. It's not very idiomatic to do `use submodule::*` and bring _everything_ into scope.

Reply:
There's a convention here that considerably reduces the reliance on glob imports and makes the code more readable but I may have to write that down to explain.

Reply:
Ah I think I saw this, where you're always prefixing with `gen`. I like that. I still see plenty of `use crate::*` that I'm not too thrilled about, but I like the strategy of using the `gen` prefix`